doxia-integration-tools/src/main/java/org/apache/maven/doxia/tools/DefaultSiteTool.java

```diff
@@ -174,20 +173,34 @@ public Artifact getSkinArtifactFromRepository( ArtifactRepository localRepositor
         return artifact;
     }
 
-    /** {@inheritDoc} */
     public Artifact getDefaultSkinArtifact( ArtifactRepository localRepository,
                                             List<ArtifactRepository> remoteArtifactRepositories )
         throws SiteToolException
     {
         return getSkinArtifactFromRepository( localRepository, remoteArtifactRepositories, new DecorationModel() );
     }
 
-    /** {@inheritDoc} */
+    /**
+     * This method is not implemented according to the URI specification and has many weird
+     * corner cases where it doesn't do the right thing. Please consider using a better 
+     * implemented method from a different library such as org.apache.http.client.utils.URIUtils#resolve.
+     */
+    @Deprecated
     public String getRelativePath( String to, String from )
     {
         checkNotNull( "to", to );
         checkNotNull( "from", from );
-
+        
+        if ( to.contains( ":" ) && from.contains( ":" ) )
```

Comment:
This could be dangerous/wrong. Consider this: `blabla.com:8080/sdfsd`. No scheme, but you'd still process here. Better check for `://`?

Reply:
Is that a mistake? For this to fire, both to and from have to have colons and both to and from have to differ in the substring before the colon, in which case it just returns the to string. The case that seems off is

http://www.example.com:99/foo.html
and 
http://www.example.com:77/bar.html

But is there a plausible relative URL between those two? 

Also, not all URLs have a //.

Reply:
Right, `mailto` does not. But Using `contains` paired with `lastIndexOf` would mean with a non-standard port that you substring scheme and authority, don't you?

Reply:
Yes, which is the intended behavior because if they're not the same we can't compute a relative URL.